mission-control/docs/config-db/concepts/custom-scrapers.md

```diff
@@ -0,0 +1,89 @@
+# Custom Scrapers
+
+Custom scrapers allow you to scrape from sources that are not well defined eg: scraping a file of an arbitrary structure. It is upon the user to define how the configs should be scraped from the source.
+
+```yaml title="file-scraper.yaml"
+apiVersion: configs.flanksource.com/v1
+kind: ScrapeConfig
+metadata:
+  name: file-scraper
+spec:
+  file:
+    - type: $.Config.InstanceType
+      class: $.Config.InstanceType
+      id: $.Config.InstanceId
+      path:
+        - config*.json
+        - test*.json
+```
+
+## Mapping
+
+Custom scrapers need to define the id, type & class for each items that are scraped. For example: if you're scraping a file with a JSON Array where each element on the array is to be scraped as a config item, you need to define what the id, type & config class of the items should be. This can be done using mappings.
+
+| Field             | Description                                                                      | Scheme   | Required |
+| ----------------- | -------------------------------------------------------------------------------- | -------- | -------- |
+| `id`              | Specify JSONPath to get the id for each config item                              | `string` | `true`   |
+| `type`            | Specify JSONPath to get the type for each config item                            | `string` | `true`   |
+| `class`           | Specify JSONPath to get the class for each config item. _(Defaults to `type`)_   | `string` |          |
+| `name`            | Specify JSONPath to get the name for each config item                            | `string` |          |
+| `createFields`    | Specify JSONPath to get the creation timestamp for each config item              | `string` |          |
+| `deleteFields`    | Specify JSONPath to get the deletion timestamp for each config item              | `string` |          |
+| `timestampFormat` | Specify the format of the timestamp specified in `createFields` & `deleteFields` | `string` |          |
+
+### Date Mapping
+
+Date mapping allows you to specify custom creation and deletion times for config items. This is useful when you want to import config items from an external source, and you want to preserve the creation and deletion times of the config items in the external source.
```

Comment:
Config items in Mission Control have created, updated and deleted date, for Kubernetes and Cloud resources this is  provided by the API but when ingesting custom config items these values need to be extracted from the config item. If no date mapping is provided the scrape time will be used inistead.